lib/config/error-codes/internal.json

```diff
@@ -120,6 +120,251 @@
           "class": "NotFoundError"
         }
       }
+    },
+    "plugins": {
```

Comment:
I don't think that plugin errors should belong to the `internal` domain: most of those errors are thrown when a plugin is badling configured/written, so that does not involve any internal issue.

These errors should probably be put in their own `plugins` domain.

Reply:
I putted it here because are thrown because of the internal service who checks the plugins configuration
Plus, if these errors are in their own plugins domain, how should i divide it ? what would be the subdomains?

Reply:
We discussed this in the original workshop: plugin related errors should have their own domain. 
Of course it's the core that checks and validates plugins, but errors thrown that way are `PluginImplementationError`.

About subdomains, there are at least 2 kind of errors that would be interesting to distinguish: `validation` and `runtime`.
Another strategy would be to have a different subdomain per plugin feature (hook, pipe, auth and controller), that would be another way to define subdomains. You'd still need a "validation" plugin subdomain obviously.

I'll let you analyze the different kinds of errors that can be thrown and devise subdomains accordingly.

Reply:
`validation` and `runtime` is fine to me, i'll add subdomains if needed in the future: here it seems to be sufficient